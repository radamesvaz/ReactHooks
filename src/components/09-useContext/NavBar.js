import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <Link to="/" className="navbar-brand">useContext</Link>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink exact to="/" activeClassName="active" className="nav-link" >Home</NavLink>
                        <NavLink exact to="/about" activeClassName="active" className="nav-link" >About</NavLink>
                        <NavLink exact to="login" activeClassName="active" className="nav-link" >Login</NavLink>
                    </div>
                </div>
            </div>
        </nav>
    )
}
