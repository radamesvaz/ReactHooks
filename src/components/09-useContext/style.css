body{
    background-color: black;
    color: #C724B1;
}

h1{
    color: #04d9ff ;
}