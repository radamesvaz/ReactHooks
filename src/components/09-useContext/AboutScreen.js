import React, { useContext } from 'react';
import { UserContext } from './UserContext';

import './style.css';

export const AboutScreen = () => {

    const { user, setUser} = useContext(UserContext);

    const handleLogOut = () => setUser({});

    return (
        <div>
            <h1>About Screen</h1>
            <hr />

            <pre>
                { JSON.stringify(user, null, 3) }
            </pre>

            <button
                className="btn btn-danger"
                onClick={ handleLogOut }
            >
                Logout
            </button>
        </div>
    )
}
 