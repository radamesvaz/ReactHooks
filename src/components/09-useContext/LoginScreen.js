import React, { useContext } from 'react';
import { UserContext } from './UserContext';

import './style.css';

export const LoginScreen = () => {

    const { setUser} = useContext(UserContext)
    
    const usuarioIngresado = {
        id: '007',
        name: 'Radames'
    }

    return (
        <div>
            <h1>Login Screen</h1>
            <hr />

            <button
                className="btn btn-outline-primary"
                onClick={ () => setUser(usuarioIngresado)}
            >
                Iniciar Sesi√≥n
            </button>
        </div>
    )
}
