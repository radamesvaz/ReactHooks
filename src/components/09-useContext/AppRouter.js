import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
      } from "react-router-dom";

import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
  

export const AppRouter = () => {
    return (
        <Router>
            <div className="container">

                <NavBar />
                
                <Switch>
                    <Route exact path="/" component={ HomeScreen } />
                    <Route exact path="/about" component={ AboutScreen } />
                    <Route exact path="/login" component={ LoginScreen } />


                    {/* Si no se consigue la ruta ingresada este es el componente que se va a mostrar */}
                    <Redirect to="/" />
                </Switch>


            </div>
        </Router>
    )
}
