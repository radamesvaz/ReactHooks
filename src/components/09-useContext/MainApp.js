import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import './style.css';
import { UserContext } from './UserContext';

export const MainApp = () => {

    const [user, setUser] = useState({})

    return (
            <UserContext.Provider value={{
                user,
                setUser
            }} >

                <AppRouter />

            </UserContext.Provider>
    )
}
