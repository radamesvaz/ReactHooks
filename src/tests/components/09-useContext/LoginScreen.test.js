import { mount } from 'enzyme';
import React from 'react';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Haciendo pruebas sobre el componente <LoginScreen />', () => {
    
    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen />
        </UserContext.Provider>
    )

    test('debe de mostrarse correctamente ', () => {
        
        expect( wrapper ).toMatchSnapshot()
    })

    test('debe de ejecutar el setUser con el argumento esperado ', () => {
        
        const button = wrapper.find('button');
        button.simulate('click');

        expect( setUser ).toHaveBeenCalled();
        expect( setUser ).toHaveBeenCalledWith( {
            id: '007',
            name: 'Radames'
        } )
    })
    
    
})
