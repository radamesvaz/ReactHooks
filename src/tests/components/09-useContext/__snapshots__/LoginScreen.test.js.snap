// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo pruebas sobre el componente <LoginScreen /> debe de mostrarse correctamente  1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Iniciar Sesi√≥n
  </button>
</div>
`;
